### ISSUE TITLE
ParallelExector Fetch两个名字一样的tensor，返回值异常

### ISSUE THREAD

**Issue by Meiyim** [PaddlePaddle/Paddle#16286]
如果您没有查询到相似问题，为快速解决您的提问，建立issue时请提供如下细节信息：
- 标题：简洁、精准概括您的问题，例如“Insufficient Memory xxx" ”
- 版本、环境信息：
    1）PaddlePaddle版本：请提供您的PaddlePaddle版本号，例如1.1或CommitID
1.3.0
    2）CPU:   预测若用CPU，请提供CPU型号，MKL/OpenBlas/MKLDNN/等数学库使用情况
    3）GPU：预测若用GPU，请提供GPU型号、CUDA和CUDNN版本号
P40, CUDA Capability: 61, Driver API Version: 10.0, Runtime API Version: 9.0
    4）系统环境：请您描述系统类型、版本，例如Mac OS 10.14，Python版本
          Python 3.6.5 (default, Oct 23 2018, 10:54:02)
          [GCC 4.8.2] on linux
- 训练信息
    1）单机/多机，单卡/多卡
单机单卡
    2）显存信息
    3）Operator信息
- 复现信息：如为报错，请给出复现环境、复现步骤
- 问题描述：请详细描述您的问题，同步贴出报错信息、日志、可复现的代码片段

Paralle Executor fetch两个名字一样的tensor scalar, 
预期返回值的shape为[1,] [1,]
实际返回值的shape为 [2,] [2,]

```python
import paddle
import numpy as np

import paddle.fluid  as F
import paddle.fluid.layers  as L

BS=50
DIM=128
SEQ=30
VOC=300000

train_program = F.Program()
startup_prog = F.Program()
with F.program_guard(train_program, startup_prog):
    with F.unique_name.guard():
        inputs = L.data(name='inputs', shape=[SEQ, 1], dtype='int64')
        #inputs = L.data(name='inputs', shape=[DIM], dtype='float32')
        labels = L.data(name='labels', shape=[1], dtype='int64')
        emb = L.embedding(inputs, [VOC, DIM], param_attr=F.ParamAttr(name='emb', initializer=F.initializer.XavierInitializer(fan_in=VOC, fan_out=DIM)))
        emb = L.reduce_sum(emb, dim=1)
        emb = L.softsign(emb)
        emb = L.fc(emb, DIM)
        mm = L.matmul(emb, emb, transpose_y=True) / 10.
        loss = L.softmax_with_cross_entropy(logits=mm, label=labels)
        loss = L.reduce_mean(loss)
        F.optimizer.AdamOptimizer(0.001).minimize(loss)


place = F.CUDAPlace(0)
start_exe = F.Executor(place)

#train_exe = F.Executor(place)
exec_strategy = F.ExecutionStrategy()
exec_strategy.num_threads = 1
exec_strategy.num_iteration_per_drop_scope = min(10, 1000) # important shit

build_strategy = F.BuildStrategy()
build_strategy.remove_unnecessary_lock = False

train_exe = F.ParallelExecutor(
    use_cuda=True,
    loss_name=loss.name,
    build_strategy=build_strategy,
    exec_strategy=exec_strategy,
    main_program=train_program,
    num_trainers=1,
    trainer_id=0)


start_exe.run(startup_prog)

fake_input = np.random.randint(low=0, high=300000, size=[BS, SEQ, 1]).astype(np.int64)
#fake_input = np.random.random([BS, DIM]).astype(np.float32)
fake_label = np.expand_dims(np.array(range(BS), dtype=np.int64), -1)

for i in range(10):
    print(paddle.__version__)
    print(loss.shape)
    loss_np, mm_np = train_exe.run(fetch_list=[loss.name, loss.name],
           feed={
                'inputs':  fake_input,
                'labels':  fake_label
            })
    print(loss_np, mm_np)
```
屏幕输出：
```shell
W0319 16:40:05.176899 72214 device_context.cc:263] Please NOTE: device: 0, CUDA Capability: 61, Driver API Version: 10.0, Runtime API Version: 9.0
W0319 16:40:05.176970 72214 device_context.cc:271] device: 0, cuDNN Version: 7.0.
W0319 16:40:05.176990 72214 device_context.cc:295] WARNING: device: 0. The installed Paddle is compiled with CUDNN 7.3, but CUDNN version in your machine is 7.0, which may cause serious incompatible bug. Please recompile or reinstall Paddle with compatible CUDNN version.
1.3.0
(1,)
[3.9096315 3.9096315] [3.9096315 3.9096315]
1.3.0
(1,)
[3.877936 3.877936] [3.877936 3.877936]
```

**Comment by Xreki** [comment]
这个可能是`fetch_list=[loss.name, loss.name]`出现相同名字情况下的bug。
